CommunicationLayer/CommDeviceControl/InternetConnectionService.cpp

```diff
@@ -0,0 +1,153 @@
+        channel_->DeclareQueue(queueName_);
+        channel_->BindQueue(queueName_, exchangeName_.toStdString());
+    }
+    catch (AmqpClient::ChannelException&)
+    {
+        if (channel_ == NULL)
+        {
+            qWarning() << "Connection failed, retrying in" << SLEEP_TIME_MILLISECONDS << "ms";
+            connectionRetryTimer_.start(SLEEP_TIME_MILLISECONDS);
+            return;
+        }
+        else
+        {
+            qWarning() << " InternetConnectionService: Error creating channel, Channel Exception";
+            throw;
+        }
+    }
+    catch (AmqpClient::ChannelErrorException&)
+    {
+        if (channel_ == NULL)
+        {
+            qWarning() << "Connection failed, retrying in" << SLEEP_TIME_MILLISECONDS << "ms";
+            connectionRetryTimer_.start(SLEEP_TIME_MILLISECONDS);
+            return;
+        }
+        else
+        {
+            qWarning() << " InternetConnectionService: Error creating channel, Channel Errror Exception";
+            throw;
+        }
+    }
+    catch (AmqpClient::AmqpResponseLibraryException&)
+    {
+        if (channel_ == NULL)
+        {
+            qWarning() << " InternetConnectionService: Error creating channel, Channel is unuseable; Amqp Resonse Library Exception";
+            throw;
+        }
+    }
+    catch (AmqpClient::ConnectionException&)
+    {
+        if (channel_ == NULL)
+        {
+            qWarning() << " InternetConnectionService: Error creating channel, Channel is unuseable; Connection Exception";
+            throw;
+        }
+    }
+    catch (AmqpClient::AmqpException::exception&)
+    {
+        if (channel_ == NULL)
+        {
+            qWarning() << "Connection failed, retrying in" << SLEEP_TIME_MILLISECONDS << "ms";
+            connectionRetryTimer_.start(SLEEP_TIME_MILLISECONDS);
+            return;
+        }
+        else
+        {
+            qWarning() << " InternetConnectionService: Error creating channel, Unknown Exception";
+            throw;
+        }
+    }
+
+    qDebug("Successful connection to RabbitMQ Server");
+}
+
+bool InternetConnectionService::connectToDataSource()
+{
+    try
+    {
+        setupChannel();
+    }
+    catch (AmqpClient::ChannelException&)
+    {
+        qWarning() << "Channel could not be created exiting program";
+        exit(EXIT_FAILURE);
+    }
+    catch (AmqpClient::ChannelErrorException&)
+    {
+        qWarning() << "Channel could not be created exiting program";
+        exit(EXIT_FAILURE);
+    }
+    catch (AmqpClient::ConnectionException&)
+    {
+        qWarning() << "Channel could not be created exiting program";
+        exit(EXIT_FAILURE);
+    }
+    catch (AmqpClient::AmqpResponseLibraryException&)
+    {
+        qWarning() << "Channel could not be created exiting program";
+        exit(EXIT_FAILURE);
+    }
+    catch (AmqpClient::AmqpException::exception&)
+    {
+        qWarning() << "Channel could not be created exiting program";
```

Comment:
If the action is going to be the same for every error, just do catch it in one statement. I see that documentation on these errors is sparse, we'll just do it like this for now. I'll fix it in the future if it becomes a problem.